File: 5sem/java/lab3/src/com/lab3/Cipher.java
package com.lab3;

public class Cipher {
    CipherStrategy strategy;

    public Cipher(CipherStrategy strategy) {
        this.strategy = strategy;
    }

    public String encrypt(String text) {
        return strategy.encrypt(text);
    }

    public String decrypt(String text) {
        return strategy.decrypt(text);
    }
}

File: 5sem/cryptography/lab1/СipherLibrary/Cipher.cs
using System;

namespace СipherLibrary
{
    public class Cipher
    {
        CipherStrategy cipherStrategy;
        public Cipher(CipherStrategy cipherStrategy)
        {
            this.cipherStrategy = cipherStrategy;
        }
        public string Encrypt(string text)
        {
            return cipherStrategy.Encrypt(text);
        }
        public string Decrypt(string text)
        {
            return cipherStrategy.Decrypt(text);
        }
    }
}
